syntax = "proto3";

package envoy.extensions.clusters.common.dns.v3;

import "udpa/annotations/status.proto";

option java_package = "io.envoyproxy.envoy.extensions.clusters.common.dns.v3";
option java_outer_classname = "DnsProto";
option java_multiple_files = true;
option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/clusters/common/dns/v3;dnsv3";
option (udpa.annotations.file_status).package_version_status = ACTIVE;

// [#protodoc-title: DNS configuration for clusters]

enum DnsLookupFamily {
  UNSPECIFIED = 0;
  AUTO = 1;
  V4_ONLY = 2;
  V6_ONLY = 3;
  V4_PREFERRED = 4;
  ALL = 5;
}
